spec/duration.html

```diff
@@ -1458,74 +1458,30 @@ <h1>
-          1. Let _untilOptions_ be OrdinaryObjectCreate(*null*).
-          1. Perform ! CreateDataPropertyOrThrow(_untilOptions_, *"largestUnit"*, *"month"*).
-          1. Let _untilResult_ be ? CalendarDateUntil(_calendarRec_, _plainRelativeTo_, _newRelativeTo_, _untilOptions_).
-          1. Let _oneYearMonths_ be _untilResult_.[[Months]].
-          1. Repeat, while abs(_months_) &ge; abs(_oneYearMonths_),
-            1. Set _months_ to _months_ - _oneYearMonths_.
-            1. Set _years_ to _years_ + _sign_.
-            1. Set _plainRelativeTo_ to _newRelativeTo_.
-            1. Set _newRelativeTo_ to ? CalendarDateAdd(_calendarRec_, _plainRelativeTo_, _oneYear_).
-            1. Set _untilOptions_ to OrdinaryObjectCreate(*null*).
-            1. Perform ! CreateDataPropertyOrThrow(_untilOptions_, *"largestUnit"*, *"month"*).
-            1. Set _untilResult_ to ? CalendarDateUntil(_calendarRec_, _plainRelativeTo_, _newRelativeTo_, _untilOptions_).
-            1. Set _oneYearMonths_ to _untilResult_.[[Months]].
-        1. Else if _largestUnit_ is *"month"*, then
-          1. Assert: CalendarMethodsRecordHasLookedUp(_calendarRec_, ~dateAdd~) is *true*.
-          1. Let _moveResult_ be ? MoveRelativeDate(_calendarRec_, _plainRelativeTo_, _oneMonth_).
-          1. Let _newRelativeTo_ be _moveResult_.[[RelativeTo]].
-          1. Let _oneMonthDays_ be _moveResult_.[[Days]].
-          1. Repeat, while abs(_days_) &ge; abs(_oneMonthDays_),
-            1. Set _days_ to _days_ - _oneMonthDays_.
-            1. Set _months_ to _months_ + _sign_.
-            1. Set _plainRelativeTo_ to _newRelativeTo_.
-            1. Set _moveResult_ to ? MoveRelativeDate(_calendarRec_, _plainRelativeTo_, _oneMonth_).
-            1. Set _newRelativeTo_ to _moveResult_.[[RelativeTo]].
-            1. Set _oneMonthDays_ to _moveResult_.[[Days]].
-        1. Else,
-          1. Assert: _largestUnit_ is *"week"*.
-          1. Assert: CalendarMethodsRecordHasLookedUp(_calendarRec_, ~dateAdd~) is *true*.
-          1. Let _moveResult_ be ? MoveRelativeDate(_calendarRec_, _plainRelativeTo_, _oneWeek_).
-          1. Let _newRelativeTo_ be _moveResult_.[[RelativeTo]].
-          1. Let _oneWeekDays_ be _moveResult_.[[Days]].
-          1. Repeat, while abs(_days_) &ge; abs(_oneWeekDays_),
-            1. Set _days_ to _days_ - _oneWeekDays_.
-            1. Set _weeks_ to _weeks_ + _sign_.
-            1. Set _plainRelativeTo_ to _newRelativeTo_.
-            1. Set _moveResult_ to ? MoveRelativeDate(_calendarRec_, _plainRelativeTo_, _oneWeek_).
-            1. Set _newRelativeTo_ to _moveResult_.[[RelativeTo]].
-            1. Set _oneWeekDays_ to _moveResult_.[[Days]].
-        1. Return ! CreateDateDurationRecord(_years_, _months_, _weeks_, _days_).
+          1. Assert: If _years_ &ne; 0 or _months_ &ne; 0, CalendarMethodsRecordHasLookedUp(_calendarRec_, ~dateAdd~) is *true*.
```

Comment:
I think this assertion and its repeats can also be moved up to unconditionally join line 1461, right?

Reply:
> I think this assertion and its repeats can also be moved up to unconditionally join line 1461, right?

And also its analogues for largestUnit of "month" or "week".